### ISSUE TITLE
NodejsToolsTestDiscoverer throws Method not found: Microsoft.IO.Path.GetFileName(System.ReadOnlySpan`1)' with VS2022 17.3

### ISSUE THREAD

**Issue by fforjan** [microsoft/nodejstools#2407]
See also https://developercommunity.visualstudio.com/t/NodejsToolsTestDiscoverer-throws-Method-/10118168

##### Expected Behavior 
Test should be discovered
##### Actual Behavior 
Exception when discovering

![image](https://user-images.githubusercontent.com/601810/184144009-85e6a507-0e72-4b23-9bb0-e6ba4ad2a321.png)

- NTVS Version: one provided with VS 17.3 
- Visual Studio Version:  Version 17.3.0
- Node.js Version: v16.14.2

##### Steps to Reproduce
simply create a nodejs app and add a mocha unit tests

**Comment by fforjan** [comment]
Ping @joj  and @safacero  as you are the one who did recent change.
This is blocking our team to migrate to VS2022 17.3

**Comment by joj** [comment]
@fforjan we'll look into it

**Comment by safacero** [comment]
@fforjan I'm not successful at repro'ing it! Can you share the exact steps with the exact project type and how you added the test and selected the Mocha framework? Thank you so much for your patience :)

**Comment by fforjan** [comment]
[NodejsConsoleApp2.zip](https://github.com/microsoft/nodejstools/files/9312263/NodejsConsoleApp2.zip)

Just open the solution, wait for the test to be discovered, check the output for the test :

![image](https://user-images.githubusercontent.com/601810/184246904-d5c7c1b1-c4f4-4a35-a89b-93df4825d9d9.png)

@safacero  let me know if you need more info

**Comment by fforjan** [comment]
(this was simply created by doing new project + add new mocha unit test)

**Comment by fforjan** [comment]
@safacero  also, look at https://developercommunity.visualstudio.com/t/Building-Xamarin-projects-stopped-workin/10097536#T-N10097747-N10098939 I would think this is related

**Comment by joj** [comment]
@fforjan do you have C++ installed in that VS installation? If you do, can you try uninstalling "Test Adapter for Boost Test" and "Test Adapter for Google Test" in Individual Components? That issue has been fixed already, but I don't quite remember what version the fix went in. Thanks! J

**Comment by fforjan** [comment]
@joj  I will quickly check but just a remark, we are talking about the latest Visual Studio 2022 which got delivered this week. If the issue is fixed, when do you think it will be available into Visual Studio 2022 update ?

**Comment by joj** [comment]
I'm out for a few days but I can check when I get back :)

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by joj** [comment]
I checked! The change was inserted into 17.4 P1. It's already available in the Preview versions. We didn't want to ship to 17.3 because we discovered it very late in the process and when we discovered the issue, it was only affecting JS tests, for people with the C++ workload installed (so, exactly this case). We didn't think many people would be using those two at the same time, which makes the workaround fairly straightforward.
